app/ui/src/Components.elm

```diff
@@ -0,0 +1,33 @@
+module Components exposing (..)
+
+import Html exposing (..)
+import Html.Attributes exposing (..)
+import Html.Events exposing (onClick, onInput, onSubmit)
+import Model exposing (Output(..))
+
+
+layout : List (Html msg) -> Html msg
+layout children =
+    div [] children
+
+
+query : (String -> msg) -> msg -> Html msg
+query queryChange submit =
+    Html.form [ onSubmit submit ]
+        [ label [ for "query" ] [ text "Your query: " ]
+        , input [ id "query", onInput queryChange ] []
+        , button [ type_ "submit", onClick submit ] [ text "Update" ]
+        ]
+
+
+output : Output -> Html msg
```

Comment:
I don't see how to achieve the same behaviour without it. Behaviour being:

> if the output is a single string, render it as a `p` (the most basic message view)
> if the output is a list of strings, render it as a `ul` (the most basic content view)

Making it just a list of string, I'll lose the ability to distinguish between a status message and a server response with a single message, which is what I want.